bundles/org.eclipse.e4.ui.css.swt/src/org/eclipse/e4/ui/css/swt/dom/ToolItemElement.java

```diff
@@ -27,14 +30,50 @@
  *
  */
 public class ToolItemElement extends ItemElement {
+
+	private boolean isSelected = false;
+
+	boolean dynamicEnabled = Boolean.getBoolean("org.eclipse.e4.ui.css.dynamic");
+
+	private SelectionListener selectionListener = new SelectionAdapter() {
+		@Override
+		public void widgetSelected(SelectionEvent e) {
+			if (!e.widget.isDisposed()) {
```

Comment:
There's no need to check for disposed in an event handler, events aren't delivered to disposed widgets. (Otherwise it's a bug in SWT).

Reply:
Thanks, updated